### ISSUE TITLE
Sensor data not correlating between switch output and librenms | Arista EOS

### ISSUE THREAD

**Issue by jsdurling** [librenms/librenms#4130]
# nms0 # ./validate.php 

Version info:
Commit SHA: 9a66326fcf90aebf9835fd846bb6e760337d3f15
DB Schema: 128
PHP: 5.4.16
MySQL: 5.5.44-MariaDB
RRDTool: 1.4.8
SNMP: NET-SNMP version: 5.7.2
# 

[OK]      Database connection successful

Attached the poller debug output.
[poller.php.zip](https://github.com/librenms/librenms/files/427650/poller.php.zip)

Arista EOS switch reporting:
```
**************************\* 556. row ***************************
                sensor_id: 6470
           sensor_deleted: 0
             sensor_class: power
                device_id: 548
              poller_type: snmp
               sensor_oid: .1.3.6.1.2.1.99.1.1.1.4.100352213
             sensor_index: 100352213
              sensor_type: entity-sensor
             sensor_descr: DOM RX Power  for Ethernet52
           sensor_divisor: 10000000
        sensor_multiplier: 1
           sensor_current: 0.0005308
             sensor_limit: 0.0007962
        sensor_limit_warn: NULL
         sensor_limit_low: NULL
    sensor_limit_low_warn: NULL
             sensor_alert: 1
            sensor_custom: No
         entPhysicalIndex: NULL
entPhysicalIndex_measured: NULL
               lastupdate: 2016-08-19 14:06:41
              sensor_prev: 0.0005307
556 rows in set (0.01 sec)

sw0.b03-01-10#show int e 52 transceiver 
If device is externally calibrated, only calibrated values are printed.
N/A: not applicable, Tx: transmit, Rx: receive.
mA: milliamperes, dBm: decibels (milliwatts).
                               Bias      Optical   Optical  
          Temp       Voltage   Current   Tx Power  Rx Power  
Port      (Celsius)  (Volts)   (mA)      (dBm)     (dBm)     Last Update  

---

Et52       28.84      3.33      5.53     -2.12     -2.75     0:00:09 ago
```
Specifically, this is just the Tx Power  Rx Power outputs.  The are showing up as sensor_current: 0.0005308

So, for "DOM RX Power  for Ethernet52" librenms is reporting "sensor_current: 0.0005308", and the actual switch output is "-2.75" . 

Any info you can provide on resolving this would be appreciated.

Thanks,
Jeff Durling

**Comment by crcro** [comment]
@jsdurling can you get the following oids values for that interface sensor:

entPhySensorType
entPhySensorScale
entPhySensorPrecision

or a full snmpwalk?

**Comment by jsdurling** [comment]
I apologize for the delay, I didn't see the response until this morning.  

Attached is the full snmpwalk on an Arista switch having the issues.

[snmpwalk-arista-eos.gz](https://github.com/librenms/librenms/files/467516/snmpwalk-arista-eos.gz)

**Comment by laf** [comment]
Can you provide the output `snmpbulkwalk -OUneb -v2c -c COMMUNITY HOSTNAME .`

**Comment by jsdurling** [comment]
Here is the bulkwalk on the switch.  Thank you!

[snmpbulkwalk-arista-eos.gz](https://github.com/librenms/librenms/files/541939/snmpbulkwalk-arista-eos.gz)

**Comment by laf** [comment]
At present we are getting the info from ENTITY-SENSOR-MIB but ARISTA-ENTITY-SENSOR-MIB.

It would be good to get the snmpwalk done with `.1.3.6.1.4.1.30065` instead of `.`

**Comment by jsdurling** [comment]
Thank you for following up.  

Here is the updated snmpwalk and snmpbulkwalk with .1.3.6.1.4.1.30065 instead of .

[snmpbulkwalk-arista-eos-30065.txt.gz](https://github.com/librenms/librenms/files/708439/snmpbulkwalk-arista-eos-30065.txt.gz)
[snmpwalk-arista-eos-30065.txt.gz](https://github.com/librenms/librenms/files/708438/snmpwalk-arista-eos-30065.txt.gz)

**Comment by laf** [comment]
Is it possible that this is a firmware bug? Arista just utilise the standard ENTITY-SENSOR-MIB so not really sure we can do much about it.

Be interesting to see if any other Arista users have issues with this.

**Comment by laf** [comment]
Fixed in #5773
